uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/leave-request-api.js

```diff
@@ -183,6 +183,33 @@ define([
         },
 
         /**
+         * Get leave request for the given id
+         *
+         * @param {object} id - leave request id
+         *
+         * @return {Promise} resolves with {Object}
+         */
+        find: function (id) {
+          $log.debug('LeaveRequestAPI.find');
+          var deferred = $q.defer();
+          var params = { id: id };
+
+          this.sendGET('LeaveRequest', 'get', params)
+            .then(function (response) {
+              if (response.values.length > 0) {
+                deferred.resolve(response.values[0]);
+              } else {
+                deferred.reject('LeaveRequest not found with this ID');
+              }
+            })
+            .catch(function (errorMsg) {
+              deferred.reject(errorMsg);
```

Comment:
Yes, but as we are returning `deferred.promise` and not the promise returned by api call, we need to reject it from here.